packages/marko-web-theme-monorail/components/blocks/primary-image.marko

```diff
@@ -62,14 +80,19 @@ $ if (image.id) imageAttrs["data-image-id"] = image.id;
       <marko-web-image-display-name block-name=blockName obj=image />
     </if>
     <div class=`${blockName}__wrapper`>
-      <marko-web-img
-        src=src
-        srcset=srcset
-        alt=image.alt
-        class=`${blockName}__image`
-        attrs=imageAttrs
-        lazyload=lazyload
-      />
+      <marko-web-picture>
+        <for|s| of=sources>
+          <@source ...s />
```

Comment:
This should be explicitly defined, not spread, in order to pass argument validation. If the required arguments for the picture source component change in the future, this should now error because the wrong props are being sent.
```suggestion
          <@source srcset=s.srcset media=s.media />
```